Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -1170,19 +1170,21 @@ extension BrowserViewController: TabManagerDelegate {
         }
 
         removeAllBars()
-        urlBar.currentURL = selected?.displayURL
         if let bars = selected?.bars {
             for bar in bars {
                 showBar(bar, animated: true)
             }
         }
 
-        toolbar?.updateBackStatus(selected?.canGoBack ?? false)
-        toolbar?.updateForwardStatus(selected?.canGoForward ?? false)
-        toolbar?.updateReloadStatus(selected?.loading ?? false)
+        if let tab = selected,
+               webView = tab.webView {
+            updateNavigationToolbarStates(tab, webView: webView)
```

Comment:
I wonder if this call could go inside the `if let webView = selected?.webView {` block above. Looks like it's checking the same thing, no?